R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+#' @param table_range a table_range given by string. For example, "A1:B2"
+#' @param sheet a tidyxl data frame.
+
+string_range_to_filter_vars <- function(sheet, table_range) {
+  cell_ref_df <- tibble::as_tibble(cellranger::as.cell_limits(table_range))
+
+  table_range_df <-
+    cell_ref_df[, 1:2] %>%
+    purrr::set_names(c("min", "max")) %>%
+    dplyr::mutate(dimension = c("row", "col")) %>%
+    tidyr::gather(key, value, -dimension) %>%
+    tidyr::unite(label, key, dimension, sep = "_") %>%
+    tidyr::spread(label, value)
+
+  string_filter_name <- rlang::sym(table_range %>% stringr::str_remove("\\:") %>% paste0("flt_", .))
+
+  data_sheet <-
+    sheet %>%
+    dplyr::mutate(!!string_filter_name :=
+      row >= table_range_df$min_row[1] &
+        row <= table_range_df$max_row[1] &
+        col >= table_range_df$min_col[1] &
+        col <= table_range_df$max_col[1])
+}
+
+#' Get the corner references of a tidyxl data frame.
+#'
+#' This is an internal function that takes a tidyxl data frame a returns a dataframe indicating the maximum and minimum corners.
+#'
+#' @param sheet a tidyxl data frame.
+
+get_corner_cell_refs <- function(sheet) {
+
+  # Automatic producedure
+  ref_df <-
+    sheet %>%
+    dplyr::summarise(
+      min_row = min(row), max_row = max(row),
+      min_col = min(col), max_col = max(col)
+    )
+  return(ref_df)
+}
+
+#' This is an internal function to wrap `enhead`. It takes header cells, data cells and a direction and combines them into a tidy data frame.
+#'
+#' @param header_data a data frame containing header cells.
+#' @param direction direction suplied to enhead.
+#' @param values the data values.
+
+enhead_tabledata <- function(header_data, direction, values = tabledata) {
+  unpivotr::enhead(
+    data_cells = values,
+    header_cells = header_data,
+    direction = direction
+  )
+}
+
+#' The the current names of header groups.
+#'
+#' This is an internal function that returns the names of header groups.
+#'
+#' @param labels list of labels.
+#' @param regex_term prefix for col_header.
+
+get_header_index <- function(labels, regex_term = "^col_header") {
+  labels %>%
+    .[stringr::str_detect(., regex_term)] %>%
+    stringr::str_remove_all("[a-z]+|[:punct:]+") %>%
+    as.numeric() %>%
+    max(., na.rm = TRUE) %>%
+    ifelse(is.finite(.), ., 0) %>%
+    `+`(1)
+}
+
+#' An ordered set of excel column indexes.
+#'
+#' A character vector containing and ordered set of excel column indexes
+#'
+#' @format A character vector of length 702
+"cols_index"
+
+#' Matches directions to unicode arrows for an interative chart
+#'
+#' A data frame with .direction, .arrow and .rate columns
+#'
+#' @format a dataframe
+"direction_plot_interactive"
+
+#' Matches directions to unicode arrows for an interative chart
+#'
+#' A data frame with .direction, .arrow and .rate columns
+#'
+#' @format a dataframe
+"direction_plot_noninteractive"
```

Comment:
Would benefit from an explanation why this set is different and has different rotation.

Reply:
Directions eventually need refactoring to `up-left` etc., but now probably isn't the time!

Reply:
> Directions eventually need refactoring to `up-left` etc., but now probably isn't the time!

Yes! I need to get around to this. Hopefully will be fairly straightforward.

Reply:
> Would benefit from an explanation why this set is different and has different rotation.

Addressed in 791d4e0ae6d9157abd02ec3bbdb8c8bf58399e9d.